tianshou/policy/modelfree/iqn.py

```diff
@@ -1,13 +1,22 @@
-from typing import Any, Dict, Optional, Union
+from typing import Any, Dict, Optional, Union, cast
 
 import numpy as np
 import torch
 import torch.nn.functional as F
 
 from tianshou.data import Batch, to_numpy
+from tianshou.data.batch import BatchProtocol
+from tianshou.data.types import ModelOutputBatchProtocol, RolloutBatchProtocol
 from tianshou.policy import QRDQNPolicy
 
 
+class ModelOutputTauBatchProtocol(ModelOutputBatchProtocol):
+    """Model outputs and taus."""
+
+    # taus for quantile regression, see https://arxiv.org/abs/1806.06923
+    taus: torch.Tensor
+
```

Comment:
move?

Reply:
I was a bit unsure on how to handle such protocols. This one is very specific to the model, so I thought that it's better to keep it here instead of moving to the more general ones

Reply:
In the end I moved them all but renamed some and added more explicit docstrings for how they should be used